### PR TITLE
igxGrid Row selection

### LINKED ISSUE #812: igxGrid row selection
Implement row selection in igxGrid.

### Research 
- [x] [Research document](https://infragisticsinc297.sharepoint.com/:w:/r/Groups/ProdDev/_layouts/15/doc.aspx?sourcedoc=%7B63740d66-6ec6-4b84-bb54-ea9cf813902a%7D&action=edit&uid=%7B63740D66-6EC6-4B84-BB54-EA9CF813902A%7D&ListItemId=3882&ListId=%7BED8CDA43-BB2D-4B0E-8ABD-5C7F4F99DC28%7D&odsp=1&env=prod)
- [x] [Technical specification](https://github.com/IgniteUI/igniteui-angular/wiki/Row-Selection-(Grid-feature))
- [x] POC

### Development
- [x] POC polish
- [x] Create row selection
- [x] Create header checkbox selection
- [x] Change cell selection
- [x] Filtering integration
- [x] Paging integration
- [x] Sorting integration
- [x] Pinning integration
- [x] Aria Support
- [x] Keyboard Navigation
- [x] API according to spec
- [x] Styling & UI implementation
- [x] Unit tests

### Documentation
- [x] ReadMe
- [x] DocFX
- [x] Sample creation
- [x] Changelog update

### Links
- development branch (https://github.com/IgniteUI/igniteui-angular/tree/nalipiev/row-selection)
- specification (https://github.com/IgniteUI/igniteui-angular/wiki/Row-Selection-(Grid-feature))

### PR DESCRIPTION
**Pull request by Lipata**
Closes #812.  

Additional information related to this pull request:
Implementation, tests and demos

### CODE REVIEW

**Comment by Lipata** [comment]
@desig9stein You can use the Grid Selection or Grid Performance samples from demos. Thank you!

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,20 @@
 
 All notable changes for each version of this project will be documented in this file.
 ## 5.3.0
+- Added `rowSelectable` property to `igxGrid`
+    - Setting `rowSelectable` to `true` enables multiple row selection for the `igx-grid` component. Adds a checkbox column that allows (de)selection of one, multiple or all (via header checkbox) rows.
+    - Selection is persistent through data virtualization, paging and all data operations (filtering, sorting).
+    - Added methods for (de)selection of rows in a `igx-grid`:
```
Comment (kdinev) [inline CHANGELOG.md]: Just provide a link to the readme of the feature. No need to describe the API here.

File: src/grid/grid.component.ts [hunk 2]
```diff
@@ -116,6 +125,17 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this.cdr.markForCheck();
     }
 
+    get filteredData() {
+        return this._filteredData;
+    }
+
+    set filteredData(value) {
+        if (this.rowSelectable) {
+            this._filteredData = value;
```
Comment (kdinev) [inline src/grid/grid.component.ts]: Why is the `_filteredData` set only when there is row selection? The update of the header checkbox should happen on that condition, but the setter should work regardless.

File: CHANGELOG.md (hunk 1 above)
Comment (ViktorSlavov) [reply CHANGELOG.md]: Thanks for clarifying! I've added a link to the readme file and the (not yet available) documentation page in the latest commit.

**Pull request merged**